src/rgw/rgw_process.cc

```diff
@@ -87,12 +90,83 @@ void RGWProcess::RGWWQ::_process(RGWRequest *req, ThreadPool::TPHandle &) {
   process->req_throttle.put(1);
   perfcounter->inc(l_rgw_qactive, -1);
 }
+bool rate_limit(rgw::sal::Store* store, req_state* s) {
+  // we dont want to limit health check or system or admin requests
+  const auto& is_admin_or_system = s->user->get_info();
+  if (s->op_type ==  RGW_OP_GET_HEALTH_CHECK || is_admin_or_system.admin || is_admin_or_system.system)
+    return false;
+  std::string userfind;
```

Comment:
why do you need this temporary variable?
can you just set the value to `s->ratelimit_user_name`?